src/dl.cpp

```diff
@@ -247,18 +247,78 @@ static dl_error_t dl_internal_store_array( dl_ctx_t dl_ctx, dl_type_storage_t st
 					return err;
 			}
 			break;
+		case DL_TYPE_STORAGE_ANYPTR:
+			for( uint32_t elem = 0; elem < count; ++elem )
+			{
+				dl_typeid_t tid = *reinterpret_cast<const dl_typeid_t*>( instance + ( ( elem * 2 + 1 ) * sizeof( void* ) ) );
+				sub_type = dl_internal_find_type( dl_ctx, tid );
+				dl_error_t err = dl_internal_store_ptr( dl_ctx, instance + ( elem * sizeof( void* ) * 2 ), sub_type, store_ctx );
+				if( DL_ERROR_OK != err )
+					return err;
+				dl_binary_writer_write_uint32( &store_ctx->writer, tid );
+				if (sizeof(uintptr_t) == 8)
+					dl_binary_writer_write_zero( &store_ctx->writer, 4 );
+			}
+			break;
 		default: // default is a standard pod-type
 			dl_binary_writer_write( &store_ctx->writer, instance, count * size );
 			break;
 	}
 	return DL_ERROR_OK;
 }
 
-static dl_error_t dl_internal_store_member( dl_ctx_t dl_ctx, const dl_member_desc* member, uint8_t* instance, CDLBinStoreContext* store_ctx )
+static dl_error_t dl_internal_store_anyarray( dl_ctx_t dl_ctx, const uint8_t* data_ptr, uint32_t count, CDLBinStoreContext* store_ctx )
```

Comment:
I'm not sure, but if you split the DL_STORAGE_TYPE mentioned above into just using _ANY and use the atom type as well I think this might turn out a bit cleaner?

Reply:
Explained here https://github.com/wc-duck/datalibrary/pull/164#discussion_r1012206239